src/components/ContentSidebar/SidebarUtils.js

```diff
@@ -95,7 +96,36 @@ class SidebarUtils {
         props: ContentSidebarProps,
         file?: BoxItem,
     ): boolean {
-        return !!file && !!props.hasSkills && hasSkillsData(file);
+        return (
+            !!file &&
+            SidebarUtils.canHaveSkillsSidebar(props) &&
+            hasSkillsData(file)
+        );
+    }
+
+    /**
+     * Determines if we should bother rendering the metadata sidebar.
+     * Relies on props and metadata data and feature enabled or not.
+     *
+     * @private
+     * @param {ContentSidebarProps} props - User passed in props
+     * @param {Array<MetadataEditor>} editors - metadata editors
+     * @param {Boolean} isMetadataEnabled - metadata feature
+     * @return {Boolean} true if we should render
+     */
+    static shouldRenderMetadataSidebar(
+        props: ContentSidebarProps,
+        editors?: Array<MetadataEditor>,
+    ): boolean {
+        const { metadataSidebarProps = {} }: ContentSidebarProps = props;
+        const {
+            isFeatureEnabled = true,
+        }: MetadataSidebarProps = metadataSidebarProps;
+
+        return (
+            SidebarUtils.canHaveMetadataSidebar(props) &&
+            (isFeatureEnabled || (Array.isArray(editors) && editors.length > 0))
```

Comment:
Pretty difficult to read with the nested conditions. Consider breaking them out.